sdks/python/apache_beam/runners/portability/flink_runner_test.py

```diff
@@ -211,6 +211,9 @@ def process(self, v):
                 counter_name, line)
         )
 
+    def test_sdf(self):
+      raise unittest.SkipTest("BEAM-2939")
```

Comment:
What is missing in the Flink Runner to support this? SDF execution should be transparent, apart from dynamic splitting and checkpointing of the residual elements which both require Runner interaction.

Reply:
Flink simply drops anything in the residual_roots field in ProcessBundleResponse https://github.com/apache/beam/blob/release-2.10.0/model/fn-execution/src/main/proto/beam_fn_api.proto#L290 . These need to be replayed back to the bundle to process the remainder.

Reply:
Just saw that the response is passed to `BundleProgressHandler` which so far only handles metric requests.